assets/js/search.js

```diff
@@ -103,99 +107,87 @@
+    } else {
+      searchContent = "Not Valid";
+    }
+    return searchContent;
+  }
+
 
   /**
-   * A function to search the IaC Lib and Deployment guides. Can also be used for other pages that need it.
+   * A function to search the Deployment guides.
    * To show/hide the proper elements based on the results.
    * @type {Function}
    */
-  function filterData(searchValue, type) {
-    const searchEntry = detectSearchEntry();
+  function filterSearchData(searchValue, type) {
+    let matches = 0;
+    let submoduleMatches = 0;
 
-    $('#guide-listings').show();
+    const searchEntry = detectSearchEntry();
 
-    $('#no-azure-results').hide();
-    $('#no-matches').hide();
+    if($('.guide-card').length !== 0){
+      initialDisplay();
+    }
 
     if (searchValue && searchValue.length > 0) {
       const searchQueries = searchValue.toLowerCase().split(" ");
 
-      if (searchEntry.type === 'libraryEntries') {
-        $('.table-clickable-row').hide();
-      } else if (searchEntry.type === 'guideEntries') {
-        $('#search-results-count').hide();
-        $('.guide-card').hide() &&
-          $('.category-head').hide() &&
-          $('.categories ul li').hide();
-      }
+      $('.table-clickable-row'.length === 0) ? hideItems() : $('.table-clickable-row').hide();
 
-      let matches = 0;
-      let submoduleMatches = 0;
-      let searchContent;
-
-      searchEntry.entries.forEach(entry => {
+      searchEntry.forEach(entry => {
         let matchesAll = true;
 
         searchQueries.forEach(searchQuery => {
-          switch (true) {
-            case searchEntry.type === 'libraryEntries':
-              searchContent = entry.text;
-              break;
-            case searchEntry.type === 'guideEntries' && type === 'wordSearch':
-              searchContent = entry.title + entry.category + entry.content + entry.tags;
-              break;
-            case searchEntry.type === 'guideEntries' && type === 'tagSearch':
-              searchContent = entry.tags + entry.cloud;
-              break;
-            case searchEntry.type === 'guideEntries' && type === 'cloudSearch':
-              searchContent = entry.cloud;
-              break;
-            default:
-              "Not Valid"
-          }
+          const searchContent = getSearchData(entry, type);
 
           if (searchContent.indexOf(searchQuery) < 0) {
             matchesAll = false;
+            $(`#${entry.id}`).hide();
           }
         });
 
         //Checks if results were found and displays results accordingly
         if (matchesAll) {
           displayCategory(entry);
-          $("#" + entry.id).show();
+          $(`#${entry.id}`).show();
           matches++;
-          (searchEntry.type === 'libraryEntries') ? submoduleMatches += entry.num_submodules: submoduleMatches = 0;
-          return;
+          $('.table-clickable-row'.length > 0) ? submoduleMatches += entry.num_submodules: submoduleMatches = 0;
```

Comment:
Same here

Reply:
Also, ternary syntax is primarily useful as an expression that returns a value, and you're trying to pick between two possible values in a one-liner. Here, you're using it for side effects, which seems harder to read without much gain. Consider turning into normal if/else.